product_variant_sale_order_route/views/product_view.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<odoo>
+
+    <record model="ir.ui.view" id="product_normal_form_view">
+        <field name="name">product_normal_form_view</field>
+        <field name="model">product.product</field>
+        <field name="inherit_id" ref="product.product_normal_form_view" />
+        <field name="arch" type="xml">
+            <field name="route_ids" position="after">
```

Comment:
IMO just set route_ids as invisible on the product.product form (so that it won't update the product.template) and we should be fine